src/sockets/sockets.ts

```diff
@@ -2295,6 +2319,270 @@ function updateRoomMaxPlayers(number) {
   updateCurrentGamesList();
 }
 
+// Functions to manage ranked and unranked queue
+
+// Unranked queue
+type MatchMakingQueueItem = {
+  id: string;
+  user: Object;
+  timeJoinedAt: Date;
+};
+
+// For unrank and rank game
+function startTimer(RoomID,io) {
+  const room = rooms[RoomID];
+  if (room.timer <= 0) {
+    //TODO task 
+    console.log("test");
+    return;
+  }
+
+  const timerInterval = setInterval(() => {
+    room.timer--;
+    io.to(RoomID).emit('timerUpdate', room.timer);
+
+    if (room.timer <= 0) {
+      clearInterval(timerInterval);
+      io.to(RoomID).emit('timerFinished');
+    }
+  }, 1000);
+}
+
+// for self-define
+function SetTimer(RoonID,io,duration) {
+  const room = rooms[RoomID];
+  room.timer = duration;
+
+  if (room.timer <= 0) {
+    //TODO void the game
+    //FR13 - task 2
+    console.log("test");
+    return;
+  }
+
+  const timerInterval = setInterval(() => {
+    room.timer--;
+    io.to(RoomID).emit('timerUpdate', room.timer);
+
+    if (room.timer <= 0) {
+      clearInterval(timerInterval);
+      io.to(RoomID).emit('timerFinished');
+    }
+  }, 1000);
+}
```

Comment:
Overall I think this game move timer should be dependent on game phase updates. Not on socket updates!

See this in game.ts:
```
  gameMove(socket, data) {
```